docs/developer.rst

```diff
@@ -1,17 +1,65 @@
 ..
-    Copyright (c)      2022 ETH Zurich
-    Copyright (c) 2007-2017 Louisiana State University
+    Copyright (c) 2024 ETH Zurich
 
     SPDX-License-Identifier: BSL-1.0
     Distributed under the Boost Software License, Version 1.0. (See accompanying
     file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-Release procedure for pika
-==========================
+.. _develop:
 
-The current target is to produce a new (minor) release once a month.
+=======================
+Developer documentation
+=======================
 
-pika follows `Semantic Versioning <https://semver.org>`_.
+Contributing
+============
+
+Contributions are always welcome. Before starting work on larger features it's recommended to
+coordinate design ideas through GitHub issues. Existing open issues may sometimes no longer be
+useful, have already been solved, or the notes on solving a particular problem may be outdated.
+Issues that we think are easier for someone new to the project to get started with are `labeled on
+the issue tracker <https://github.com/pika-org/pika/issues/1118>`__.
+
+Building documentation
+======================
+
+Building pika's documentation requires doxygen, sphinx, as well the sphinx extensions and themes
+listed in ``docs/conf.py``. Building the documentation is not done through CMake as the rest of
+pika. Assuming ``SOURCE_DIR`` is set to the pika source directory, and ``BUILD_DIR`` is set to a
+build directory (e.g. ``$SOURCE_DIR/build``), you first need to export the following environment
+variables for doxygen:
+
+.. code-block:: bash
+
+   export PIKA_DOCS_DOXYGEN_INPUT_ROOT="$SOURCE_DIR"
+   export PIKA_DOCS_DOXYGEN_OUTPUT_DIRECTORY="$BUILD_DIR/doxygen"
+
+Then generate the doxygen XML files using:
+
+.. code-block:: bash
+
+   doxygen "$SOURCE_DIR/docs/Doxyfile"
+
+Finally, build the sphinx documentation using:
+
+.. code-block:: bash
+
+   sphinx-build --fail-on-warning --builder html "$SOURCE_DIR/docs" "$BUILD_DIR/docs"
```

Comment:
Do you have a link for this? We've been using `-W`, the short version of the option since way before 7.3.0, but maybe they added the long version only recently?

In general though, yes, it'd be good to have some notes on versions, but I actually don't know what versions we require, other than us using the latest available from `pip` in CI. We should presumably pin the dependencies, but I also kind of like using the latest versions so we know when it breaks. What do you think would be best?

Reply:
https://github.com/sphinx-doc/sphinx/pull/11776

Reply:
I changed the example to use the short options...